pkg/distributor/distributor_test.go

```diff
@@ -2698,58 +2749,78 @@ func TestDistributor_LabelNamesAndValues(t *testing.T) {
 			Values:    []string{"200", "500"},
 		},
 	}
-	tests := map[string]struct {
-		zones              []string
-		zonesResponseDelay map[string]time.Duration
```

Comment:
Note to reviewers: the test case on zone awareness only makes sense when partitions are not used (quorum works differently with partitions). So I've extracted this test case in a dedicated `t.Run()`.